### ISSUE TITLE
ResourceResolver and MessageSource look very redundant

### ISSUE THREAD

**Issue by keilw** [jsr377/jsr377-api#7]
With generic signatures in ResourceResolver, it and MessageSource feel almost identical.
It could be more efficient to have ResourceResolver< String > or at most a MessageSource (or MessageResource ;-)  simply extend ResourceResolver< String > providing extra methods like asResourceBundle().

**Comment by hendrikebbers** [comment]
+1
I think it's ok to call "String myMessage = source.resolveResource("dialog.title", myLocal)" instead of "String myMessage = source.getMessage("dialog.title", myLocal)" since a text/message is a resource, too.

**Comment by aalmiray** [comment]
True, they look very similar but `MessageSource` has a narrower contract than `ResourceResolver` (it does not need the `Converted` assisted methods). Also, `ResourceResolver` cannot be generified to a single type (see #6)
